src/main/java/org/broadinstitute/hellbender/tools/funcotator/mafOutput/MafOutputRenderer.java

```diff
@@ -0,0 +1,764 @@
+     * @param outputMap {@link Map} of {@link String} to {@link Object} to hold output annotations.  Must not be {@code null}.
+     * @param key {@link String} key to add to the output map.  Must not be {@code null}.
+     * @param value {@link Object} value for the output map.
+     */
+    private void setField(final Map<String, Object> outputMap, final String key, final Object value) {
+        Utils.nonNull(outputMap);
+        Utils.nonNull(key);
+
+        if ( value != null ) {
+            outputMap.put(key, value);
+        }
+        else {
+            outputMap.put(key, UNKNOWN_VALUE_STRING);
+        }
+    }
+
+    /**
+     * Write the given line to the {@link #printWriter}.
+     * @param line The {@link String} to write as a line to the {@link #printWriter}.
+     */
+    private void writeLine(final String line) {
+        printWriter.write(line + System.lineSeparator());
+    }
+
+    /**
+     * Write the header to the output file.
+     * @param outputMap A populated output map from which to derive the header columns.
+     */
+    protected void writeHeader(final LinkedHashMap<String, String> outputMap) {
+        // Write out version:
+        writeLine(COMMENT_STRING + "version " + VERSION);
+        writeLine(COMMENT_STRING + COMMENT_STRING);
+
+        // Write previous header info:
+        for ( final VCFHeaderLine line : inputFileHeader.getMetaDataInInputOrder() ) {
+            printWriter.write(COMMENT_STRING);
+            printWriter.write(COMMENT_STRING);
+            printWriter.write(" ");
+            writeLine( line.toString() );
+        }
+
+        // Write any default tool header lines:
+        for ( final String line : toolHeaderLines ) {
+            printWriter.write(COMMENT_STRING);
+            printWriter.write(COMMENT_STRING);
+            printWriter.write(" ");
+            writeLine(line);
+        }
+
+        // Write tool name and the data sources with versions:
+        printWriter.write(COMMENT_STRING);
+        printWriter.write(COMMENT_STRING);
+        printWriter.write(" ");
+        printWriter.write(" Funcotator ");
+        printWriter.write(Funcotator.VERSION);
+        printWriter.write(" | Date ");
+        printWriter.write(new SimpleDateFormat("yyyymmdd'T'hhmmss").format(new Date()));
+        printWriter.write(getDataSourceInfoString());
+        writeLine("");
+
+        // Write the column headers for our output set and our manual annotations:
+        printWriter.write( outputMap.keySet().stream().collect(Collectors.joining(FIELD_DELIMITER)) );
+        writeLine(FIELD_DELIMITER + manualAnnotations.keySet().stream().collect(Collectors.joining(FIELD_DELIMITER)));
+
+        // Make sure we keep track of the fact that we've now written the header:
+        hasWrittenHeader = true;
+    }
+
+    /**
+     * Initializes {@link MafOutputRenderer#defaultMap} with the default keys for the columns in a MAF file.
+     */
+    protected void initializeDefaultMapWithKeys() {
```

Comment:
Can we drive this from a config file or just a list of MAF field Strings?

Reply:
Yeah - the plan is to have a config file for funcotator that can have all these fields in it, but there needs to be a hard-coded default in the case of no config.